### ISSUE TITLE
Styling is different for the slds-card depending on its container.

### ISSUE THREAD

**Issue by erocheleau** [salesforce-ux/design-system#691]
> (REQUIRED) Include one or more screenshots if applicable, as well as a Codepen with the
reduced test case.

![image](https://user-images.githubusercontent.com/2488155/129782156-d07c6d4a-b303-4113-95ee-90cba3c2e577.png)

> Describe the issue. Is it a bug or a feature request (new component, new icon, new CSS class)?

There is a rule in salesforce-lightning-design-system css that comes shipped with both the local development server and inside Experience Cloud communities that change the styling of cards within cards. You can see in the screenshot that some of the CSS for a `.slds-card` gets overridden in many different cases namely the border is removed and the box-shadow is removed as well.

When you try to use the `<lightning-card>` LWC, you cannot override those styling issues because the CSS of your custom LWC cannot affect default LWC.

So we have no control if we actually want to display a card with its normal styling within a tab for example or within the local development server.

This is what a card looks like with **just** a `<lightning-card>` in my LWC:

![image](https://user-images.githubusercontent.com/2488155/129782869-c9e0544c-7b25-4813-9637-a26981c0d0ee.png)

This is what the card **should** look like if I disable the CSS overrides:

![image](https://user-images.githubusercontent.com/2488155/129783023-63224c37-9106-451f-b7d6-2694021c8fad.png)

> Is this issue related to a specific component, variant, and/or state? If so, please detail which.

Lightning Card

> Are any specific browsers impacted by this bug?

All browsers affected

> Which version of the Salesforce Lightning Design System are you using?

This bug happens everywhere where you are trying to use a card within a card, or a card within a modal, or a card within tabs, etc.

> What steps and/or code are needed to reproduce this issue?

```html
  <lightning-card title="Hello" class="override-card">
    <lightning-button label="New" slot="actions"></lightning-button>
    <p class="slds-p-horizontal_small">Card Body (custom component)</p>
    <p slot="footer">Card Footer</p>
  </lightning-card>
```

> What did you expect to happen?

I expected the slds-card to always look the same because I have no control over the style when forced to use a card within a tab/another card/modal/etc.

> What actually happened?

Styling is different for the slds-card depending on its container.

**Comment by alexiscordova** [comment]
Hi @erocheleau, taking over for @kflournoy-sf.

First, can you tell me what version of SLDS is being used? Looking at developer.salesforce.com, it seems they are loading 2.13.7 (Winter '21) while Styling Hooks for Cards was implemented in 2.15.1 (Summer '21). If you're using the same version, it's most likely the reason why your CSS Custom Property failed.

You are correct, however, that nested cards and cards in modals/tabs lose their `border` and `box-shadow` styles. This was an intentional decision based on the needs of an internal team. That said, we plan on GA'ing Styling Hooks in the next release (Winter '22). While we have some more work to do, with Styling Hooks in place you will be able to do something like the following (note: this is not final code):
<img width="1140" alt="Screen Shot 2021-09-01 at 10 35 47 AM" src="https://user-images.githubusercontent.com/1877250/131718038-f360bc11-4715-46f2-b5b5-6b0014cb0c59.png">

In your screenshots, however, I see only 1 `<lightning-card>` that isn't nested or in a modal or tab (from the best I can tell). In the rendered browser output, is there a `slds-card_wrapper` class? Is the browser inspector highlighting which selector combination is triggering the styling reset (in my screenshot above, `.slds-card .slds-card` is highlighted)?

**Comment by nathanlb** [comment]
Hi @alexiscordova 
I'm dealing with the same issue and I believe the card in @erocheleau's screenshot is in fact nested because the localdev lwc preview container itself is an `.slds-card`.
Are you saying that after the Winter '22 release styling hooks will no longer break in localdev?
Thanks

**Comment by alexiscordova** [comment]
@nathanlb Ah! That's unexpected but now makes more sense as to why this is happening with 1 card in localdev. Once our Styling Hooks work is in, this should no longer be an issue in production.

As for Styling Hooks not working in localdev, this LWC preview is out side of the purview of SLDS so I'm reaching out to some of our other engineering partners to see what's going on here. From the error message, it looks like `postcss-plugin-lwc` may be the culprit, but I'll find out and reply once I have an answer.

**Comment by alexiscordova** [comment]
@nathanlb @erocheleau for the issues around the LWC local development beta (specifically Styling Hooks not working and it using an old version of SLDS), please file an issue here https://github.com/forcedotcom/lwc-dev-server-feedback/issues. They should be able to assist further.

We will be working on implementing and GA'ing Styling Hooks this release, and styling nested cards is definitely on our list. Once this is in place, you will be able to style nested cards as you desire. Thanks for bringing this to our attention! Hope this helps.

**Comment by erocheleau** [comment]
@alexiscordova Sorry was unavailable for that period where you replied. @nathanlb and I work together, so yes, everything he said is indeed correct. Thanks a lot for the reply and followup!
